corporate/lib/stripe.py

```diff
@@ -1091,6 +1105,55 @@ def get_billing_page_context(self) -> Dict[str, Any]:
                 }
         return context
 
+    def do_initial_upgrade(
+        self, initial_upgrade_request: InitialUpgradeRequest
+    ) -> Tuple[Optional[str], Optional[Dict[str, Any]]]:
+        customer = self.get_customer()
+
+        if self.is_sponsored_or_pending(customer):
+            return reverse("sponsorship_request"), None
+
+        onboarding = initial_upgrade_request.onboarding
+        billing_page_url = reverse("billing_home")
+        if customer is not None and (
+            get_current_plan_by_customer(customer) is not None or onboarding
+        ):
+            if onboarding:
+                billing_page_url = f"{billing_page_url}?onboarding=true"
+            return billing_page_url, None
+
+        percent_off = Decimal(0)
+        if customer is not None and customer.default_discount is not None:
+            percent_off = customer.default_discount
+
+        exempt_from_license_number_check = (
+            customer is not None and customer.exempt_from_license_number_check
+        )
+
+        seat_count = self.current_count_for_billed_licenses()
+        signed_seat_count, salt = sign_string(str(seat_count))
+        context: Dict[str, Any] = {
+            "seat_count": seat_count,
+            "signed_seat_count": signed_seat_count,
+            "salt": salt,
+            "min_invoiced_licenses": max(seat_count, MIN_INVOICED_LICENSES),
+            "default_invoice_days_until_due": DEFAULT_INVOICE_DAYS_UNTIL_DUE,
+            "exempt_from_license_number_check": exempt_from_license_number_check,
+            "plan": "Zulip Cloud Standard",
+            "free_trial_days": settings.FREE_TRIAL_DAYS,
+            "onboarding": onboarding,
+            "page_params": {
+                "seat_count": seat_count,
+                "annual_price": 8000,
+                "monthly_price": 800,
```

Comment:
These prices should become a function of the tier. Probably a good way to do it is `self.pricing_by_tier[tier]` could be a PricingTier object containing `.name` ("Zulip Cloud Standard"), `.annual_price` and `.monthly_price`.

And then `InitialUpgradeRequest` would gain a tier ID as a parameter, which for now can be hardcoded to `CustomerPlan.STANDARD` in that function, but will become a parameter there too once we redesign /plans to support that.